### PR TITLE
[Plugins] Build and install `_CompilerPluginSupport` module directory.

### PR DESCRIPTION
**Pull request by DougGregor**
Fix the generated layout for the `_CompilerPluginSupport` Swift module, so that properly-named Swift interfaces are placed within a `.swiftmodule` subdirectory. Install that directory into lib/swift.

### CODE REVIEW

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Review by rxwei — APPROVED** [review]

File: lib/CompilerPluginSupport/CMakeLists.txt [hunk 1]
```diff
@@ -56,4 +77,7 @@ if(SWIFT_SWIFT_PARSER)
       DESTINATION "lib${LLVM_LIBDIR_SUFFIX}"
       COMPONENT compiler)
 
+    swift_install_in_component(DIRECTORY "${module_base}"
```
Comment (rxwei) [inline lib/CompilerPluginSupport/CMakeLists.txt:80]: Do we still need line 66-72?

Comment (DougGregor) [reply lib/CompilerPluginSupport/CMakeLists.txt:80]: I think that's installing the dylib itself

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test macOS

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
@swift-ci please smoke test

File: lib/CompilerPluginSupport/CMakeLists.txt (hunk 1 above)
Comment (DougGregor) [reply lib/CompilerPluginSupport/CMakeLists.txt:80]: They're both installed into `lib`, without the platform, because they're host content.

**Comment by DougGregor** [comment]
@swift-ci please smoke test

**Comment by DougGregor** [comment]
LLDB failure looks unrelated:
  lldb-api :: lang/swift/clangimporter/static_archive/TestSwiftStaticArchiveTwoSwiftmodules.py

rerunning

**Comment by DougGregor** [comment]
@swift-ci please smoke test macOS

File: lib/CompilerPluginSupport/CMakeLists.txt (hunk 1 above)
Comment (rxwei) [reply lib/CompilerPluginSupport/CMakeLists.txt:80]: Sorry I meant to say `lib/swift`. Currently the library is installed to `lib` but the module is installed to "lib${LLVM_LIBDIR_SUFFIX}/swift".

**Comment by DougGregor** [comment]
@swift-ci please smoke test macOS

**Comment by DougGregor** [comment]
@swift-ci please smoke test

File: lib/CompilerPluginSupport/CMakeLists.txt (hunk 1 above)
Comment (DougGregor) [reply lib/CompilerPluginSupport/CMakeLists.txt:80]: Both are going into `lib` with this pull request. We could decide that they go somewhere else, but there's some precedent for using `lib`: `lib_InternalSwiftScan.dylib`, `lib_InternalSwiftStaticMirror.dylib`, `libsourcekitdInProc.dylib` and `libswiftDemangle.dylib`, all host libraries, all go there.

**Comment by DougGregor** [comment]
Huh:

```
Exit Code: -9
Timeout: Reached timeout of 600 seconds

Command Output (stdout):
--
lldb version 13.0.0git (https://github.com/apple/llvm-project.git revision b2590c88ba87f8a3a3130cba4e235dfebdbb4cff)
Apple Swift version 5.8-dev (LLVM b2590c88ba87f8a, Swift 7cb5875ef98af4d)

--
Command Output (stderr):
--
warning: Swift error in scratch context: New Swift image added: /Users/ec2-user/jenkins/workspace/swift-PR-macos-smoke-test/branch-main/build/buildbot_incremental/lldb-macosx-x86_64/lldb-test-build.noindex/lang/swift/struct_change_rerun/TestSwiftStructChangeRerun.test_swift_struct_change_rerun_dsym/a.out. ClangImporter needs to be reinitialized..
Shared Swift state for a.out has developed fatal errors and is being discarded.
REPL definitions and persistent names/types will be lost.
```

**Comment by DougGregor** [comment]
@swift-ci please smoke test macOS

File: lib/CompilerPluginSupport/CMakeLists.txt (hunk 1 above)
Comment (rxwei) [reply lib/CompilerPluginSupport/CMakeLists.txt:80]: `lib` makes sense to me.

**Pull request merged**